worker-api/src/main/java/com/hpe/caf/api/worker/HashStoreResult.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2015-2017 Hewlett Packard Enterprise Development LP.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.api.worker;
+
+import java.util.Objects;
+
+/**
+ * The result of storing an asset in a DataStore when
+ * a hash of the asset is requested at the time of storing.
+ */
+public final class HashStoreResult
+{
+    /**
+     * Absolute reference to the stored data, which can be used to retrieve.
+     */
+    private String reference;
+
+    /**
+     * SHA-256 hash of the stored data.
+     */
+    private String hash;
+
+    public HashStoreResult(){}
+
+    public HashStoreResult(final String reference, final String hash)
+    {
+        this.reference = Objects.requireNonNull(reference);
+        this.hash = Objects.requireNonNull(hash);
+    }
+
+    public String getReference()
+    {
+        return reference;
+    }
+
+    public void setReference(final String reference)
```

Comment:
I don't think you need setters on this type. And you can make the reference and hash final keeping this type immutable.